AsyncDisplayKit/ASTableView.mm

```diff
@@ -903,6 +915,21 @@ - (void)rangeController:(ASRangeController *)rangeController didDeleteSectionsAt
     return; // if the asyncDataSource has become invalid while we are processing, ignore this request to avoid crashes
   }
 
+  if (_contentOffsetAdjustmentTopVisibleRow != nil) {
+    __block NSInteger adjustedSection = _contentOffsetAdjustmentTopVisibleRow.section;
+    __block NSInteger adjustedRow = _contentOffsetAdjustmentTopVisibleRow.row;
+    [indexSet enumerateIndexesUsingBlock:^(NSUInteger idx, BOOL *stop) {
+        if (adjustedSection > idx) {
+          adjustedSection--;
+        } else if (adjustedSection == idx) {
+          adjustedRow = 0;
```

Comment:
Hmmm in looking at the code I think you're right about that. I don't see you updating the actual _contentOffsetAdjustment, though to account for the height of the added or removed sections if they are before the topVisibleRow. Is that happening in a subtle way that I've missed?